### ISSUE TITLE
Rocket.Chat won't load when being run out of a sub folder

### ISSUE THREAD

**Issue by aberkvam** [RocketChat/Rocket.Chat#10155]
### Description:

Rocket.Chat won't load when being run from a sub folder in Apache (as per https://rocket.chat/docs/installation/manual-installation/running-in-a-sub-folder/ ).  This seems to be because assets are being specified with absolute references.

There are two categories of errors.  The first category is simple static URLs that start with a slash.  I was able to work around these with the following Apache RewriteRules:

```
RewriteRule ^/theme.css(.*)     http://localhost:3000/theme.css$1 [P,L]
RewriteRule ^/261b1e92228b079f56e663d94be069ae1300006d.css      http://localhost:3000/261b1e92228b079f56e663d94be069ae1300006d.css [P,L]
RewriteRule ^/85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.(.*)     http://localhost:3000/85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.$1 [P,L]
```

These seem to be caused by line 102 of programs/server/packages/rocketchat_theme.js, line 27 of programs/web.browser/program.json and programs/web.cordova/program.json, and lines 9 and 18 of programs/web.browser/program.json.

Unfortunately, the above RewriteRules just got me to a new problem. The browser console gives the error "There is no route for the path: /chat/".  (See complete logs below.) I haven't been able to debug/work around this but it seems to be related to the loading of router.js in the file programs/server/packages/kadira_flow-router.js.

### Server Setup Information:

* Version of Rocket.Chat Server: 0.62.1
* Operating System: Debian 9
* Deployment Method(snap/docker/tar/etc): snap
* Number of Running Instances: 1
* DB Replicaset Oplog: Enabled
* Node Version: 8.8.1 - x64
* mongoDB Version: [embedded in snap]

### Steps to Reproduce:

1. Configure as per https://rocket.chat/docs/installation/manual-installation/running-in-a-sub-folder/
2. Try to load the initial page from the external HTTPS URL.

### Expected behavior:

The page should load.

### Actual behavior:

The page is blank.  Errors appear in the browser console.

### Relevant logs:

```
There is no route for the path: /chat/
f._notfoundRoute @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
o @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
h.dispatch @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
h.replace @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
e._page.(anonymous function) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
h.start @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
h @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
f.initialize @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
(anonymous) @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:177
a @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:1
u @ 85b95b6c461d9ea5aa30e778b6ec375dfd1e080c.js?meteor_js_resource=true:1
```

**Comment by TwizzyDizzy** [comment]
@rocket-cat close

Hi @aberkvam 

we believe that the issue you reported has already been reported by somebody else. Please have a look at #3092 for further information. If you believe we closed your issue in error and your issue is, in fact, not the same as the one mentioned before, feel free to get back to us here and tell us why.

Cheers
Thomas

**Comment by aberkvam** [comment]
#3092 looks like it was fixed with commit ff1ba34 over a year ago.  My issue certainly seems to be related but involves different files.

**Comment by panki27** [comment]
I'm having the same problem. Running Ubuntu 16.04 with Apache, installed via Snap.
Upon accessing domain.com/chat, Assets and Stylesheets are being loaded from domain.com/ when they should be loaded from domain.com/chat
Is there a quick workaround for this?
